server/services/v1/database/sessions.go

```diff
@@ -180,8 +190,27 @@ func (sessMgr *SessionManager) Create(ctx context.Context, trackVerInOwnTxn bool
 	}
 	tenant, err := sessMgr.tenantMgr.GetTenant(ctx, namespaceForThisSession)
 	if err != nil {
-		log.Warn().Err(err).Msgf("Could not find tenant, this must not happen with right authn/authz configured")
-		return nil, errors.NotFound("Tenant %s not found", namespaceForThisSession)
+		// localize the namespace from metadata cluster.
+		if config.DefaultConfig.Auth.NamespaceLocalization.Enabled {
```

Comment:
I thought if this flag is true then it is localized i.e. it won't consult the metadata cluster.

Reply:
if localization is enabled - then it will get it from metadata cluster and apply in local cluster.